### PR TITLE
Add deps-dump feature to Lavamoat

### PR DESCRIPTION
**Pull request by EtDu**
- depsDump plugin option
- pass config option to createConfigSpy
- format module data with ID as the key
- if depsDump enabled, write module data to disk
- test added

### CODE REVIEW

**Review by kumavis — CHANGES_REQUESTED** [review]
rename the option to `writeAutoConfigDebug` and allow it to be true with a default path, or a string that is used as a path

File: src/index.js [hunk 1]
```diff
@@ -186,6 +190,12 @@ function getConfigurationFromPluginOpts (pluginOpts) {
     throw new Error('LavaMoat - Unrecognized value for writeAutoConfig')
   }
 
+  if (typeof pluginOpts.writeAutoConfigDebug === 'string') {
+    configuration.writeAutoConfigDebug = pluginOpts.writeAutoConfigDebug
+  } else if (pluginOpts.writeAutoConfigDebug !== false) {
```
Comment (kumavis) [inline src/index.js]:
```suggestion
  } else if (pluginOpts.writeAutoConfigDebug) {
```

File: src/generateConfig.js [hunk 2]
```diff
@@ -130,16 +133,27 @@ function aggregateDeps ({ packageModules, moduleIdToPackageName }) {
...
     cb(null, data)
   }, (cb) => {
     // call flush observer
     onEnd()
-    // end normally
+    // if writeAutoConfigDebug option enabled, dump allDeps to a file
+    if (writeAutoConfigDebug) {
+      const serialized = jsonStringify(allDeps, { space: 2 })
+      console.warn(`deps-dump - writing to ${writeAutoConfigDebug}`)
+      fs.writeFileSync(writeAutoConfigDebug, serialized)
```
Comment (kumavis) [inline src/generateConfig.js]: this part
```js
    // if writeAutoConfigDebug option enabled, dump allDeps to a file
    if (writeAutoConfigDebug) {
      const serialized = jsonStringify(allDeps, { space: 2 })
      console.warn(`deps-dump - writing to ${writeAutoConfigDebug}`)
      fs.writeFileSync(writeAutoConfigDebug, serialized)
    }
```

should appear in index.js or its own file. there should be a through or a spy added at the `'debug'`  pipeline phase

**Review by kumavis — CHANGES_REQUESTED** [review]

File: src/generateConfig.js (hunk 2 above)
Comment (EtDu) [reply src/generateConfig.js]: doooone

File: src/index.js [hunk 3]
```diff
@@ -1,8 +1,11 @@
 const fs = require('fs')
 const path = require('path')
+const clone = require('clone')
 const mkdirp = require('mkdirp')
+const through = require('through2')
```
Comment (kumavis) [inline src/index.js]:
```suggestion
const through = require('through2').obj
```

**Review by kumavis — CHANGES_REQUESTED** [review]

**Pull request merged**